### PR TITLE
add PhotonOS 3.0

### PR DESCRIPTION
**Pull request by bryceml**
### What does this PR do?
Adds PhotonOS 3.0 to the test suite

Note that `libsodium` and `libgit2-devel` are missing as they aren't in the photon os repos.

### CODE REVIEW

**Comment by garethgreenaway** [comment]
https://github.com/saltstack/salt/pull/58991 should address the failing tests on Photon OS.

**Comment by bryceml** [comment]
re-run full pr-photon3-py3-pytest

**Comment by garethgreenaway** [comment]
re-run all

**Review by ScriptAutomate — DISMISSED** [review]

File: tests/integration/modules/test_ssh.py [hunk 1]
```diff
@@ -128,6 +128,9 @@ def test_get_known_host_entries(self):
             )
 
     @pytest.mark.slow_test
+    @pytest.mark.skipif(
+        salt.utils.platform.is_photonos() is True, reason="Skip on PhotonOS"
```
Comment (waynew) [inline tests/integration/modules/test_ssh.py:132]: **question** can we say *why* we need to skip on photon here?

Comment (waynew) [reply tests/integration/modules/test_ssh.py:132]: (same for the rest of these)

**Review by waynew — CHANGES_REQUESTED** [review]
Just some questions around skipping tests

File: tests/integration/modules/test_ssh.py (hunk 1 above)
Comment (garethgreenaway) [reply tests/integration/modules/test_ssh.py:132]: A lot of these will likely be removed once we figure out the SSH key thing.

**Comment by Ch3LL** [comment]
Looks like some remaining tests and a merge conflict needing resolve are left on this one.

**Comment by garethgreenaway** [comment]
This one is super old and outdated, closing it out in favor of  #61500

**Pull request closed without merging**